### ISSUE TITLE
Release package's Eclipse does not handle *.s on case-sensitive file systems

### ISSUE THREAD

**Issue by whiterabbit963** [foss-for-synopsys-dwc-arc-processors/toolchain#113]
This issue is well described with the following bug report.  It also provides a few different solutions and workarounds.
https://bugs.eclipse.org/bugs/show_bug.cgi?id=278026

**Comment by anthony-kolesov** [comment]
Hi @douglaspeer ,

As far as I see, it is a 12 year old bug in Eclipse (https://bugs.eclipse.org/bugs/show_bug.cgi?id=105022), not in ARC-specific code. In ARC plugins we already recognize both .s and .S: https://github.com/foss-for-synopsys-dwc-arc-processors/arc_gnu_eclipse/blob/master/org.eclipse.cdt.cross.arc.gnu/plugin.xml#L28
Is there anything that we can do about this in our code?

Anton

**Comment by whiterabbit963** [comment]
@anthony-kolesov I was using the Eclipse IDE (Mars) packaged in the latest release.  Does this setup not include the ARC plugins? Also, after glancing at the ARC plugins, it appears to me the issue still exists with those as well since the problem is due to code doing case-insensitive string matching.

If the IDE was an "off-the-shelf" version, I would agree with you.  Since it is packaged with your release bundle, however, it would be nice if this just worked out of the box.  The issue is also very easy to reproduce with a new project.

While I don't pretend to understand how to extend/modify eclipse, is there a technical reason why this cannot be accomplished in your release package?

**Comment by anthony-kolesov** [comment]
Hi @douglaspeer ,

As far as I see from the comments at the https://bugs.eclipse.org/bugs/show_bug.cgi?id=278026 , there is only one proposed solution, in the patch: https://bugs.eclipse.org/bugs/attachment.cgi?id=164154&action=edit . But that means modifying generic Eclipse (not even the CDT). We don't modify Eclipse generic code in ARC GNU plugins. If there is a bug in Eclipse - it is up to Eclipse community to provide a patch. If they don't feel like solution is right, then it's their call. If you can pinpoint to an exact solution that can be done in a plugin that runs on top of current Eclipse - we will look into integrating it, but that patch in the bugreport doesn't look applicable to our case, IMHO. 

Our "IDE" is an actual off-the-shelf Eclipse, it Eclipse+CDT  downloaded from their site, we only install ARC plugins and add prebuilt ARC toolchain, then it is packed back into a single archive, for the convenience for ease of deployment for those who want a one click solution (this might not work for all users). ARC plugins are also provided standalone and can be installed by the user in their own compatible version of Eclipse (for example ASIP tools from Synopsys) - resulting Eclipse would be same as in our package. It is not a custom IDE designed on top of Eclipse, but merely a collection of multiple software packages (Eclipse, ARC plugins, OpenOCD, GNU toolchains for ARC).

**Comment by whiterabbit963** [comment]
@anthony-kolesov 
Ok, I think I understand the problem more clearly now.  The proposed patch does solve the immediate problem (I tried it).  According to https://bugs.eclipse.org/bugs/show_bug.cgi?id=105022 it does have unintended consequences with other plugins that assume case-insensitivity (maybe even your ARC plugins included).  They also talk about a solution that takes case-sensitivity flags in the plugin.xml, which seems like considerably more work than a one-line patch.

Out of my own curiosity, I edited the binary file arc_gnu_2016.09/eclipse/configuration/org.eclipse.core.runtime/.mainData.2 where it has the ASCII "S,s,asm,ASM" for the file-extension type.  This string correlates to the file-extension string named "asmSource" in the file org.eclipse.cdt/core/corg.eclipse.cdt.core/plugin.xml.  I swapped the position of the "S,s" to read "s,S" and my test build started working with .s files and failed with .S files.  I created the project as an ARC c project.

As for how to tell the ARC plugin to override the default cdt behavior, I do not know how that is to be done.
